AK/BinarySearch.h

```diff
@@ -65,8 +65,39 @@ constexpr auto binary_search(
     return nullptr;
 }
 
+// Unlike their std equivalents, these two function require the entire Container to be sorted!
+// std::[lower,upper]_bound only require the array to be sorted after and before, respectively, the needle.
+
+template<typename Container, typename Needle, typename Comparator = DefaultComparator, typename Return = decltype(&Container {}[0])>
+constexpr auto lower_bound(
+    Container&& haystack,
+    Needle&& needle,
+    Comparator comparator = Comparator {}) -> size_t
+{
+    size_t index {};
+    binary_search(haystack, needle, &index, comparator);
+    if (index > 0 && haystack[index] >= needle)
+        index -= 1;
```

Comment:
won't this do the wrong this for e.g. `lower_bound({1, 2, 2, 2, 2, 2, 2, 3}, 2)`? binary_search will return a 2 in the middle, and then you only go one left at most, when you need to go quite a bit to the left.

Reply:
Absolutely, I tried to be lazy and reuse the binary search but didn't realize it wouldn't work. Is that the issue which made the CI fail? Isn't it correct to assume that `m_boundaries` is monotonically increasing?

Reply:
I think reusing the binary search is a good idea, but I think we need to change it to make sure it always finds the leftmost element if there are multiple identical elements.

Reply:
…and I would've expected that `m_boundaries` is monotonically increasing.